loader/loader_test.go

```diff
@@ -0,0 +1,423 @@
+package loader
+
+import (
+	"fmt"
+	ecapiv1alpha1 "github.com/hacbs-contract/enterprise-contract-controller/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/tekton"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"strings"
+)
+
+var _ = Describe("Release Adapter", Ordered, func() {
+	var (
+		loader          ObjectLoader
+		createResources func()
+		deleteResources func()
+
+		application                *applicationapiv1alpha1.Application
+		component                  *applicationapiv1alpha1.Component
+		enterpriseContractPolicy   *ecapiv1alpha1.EnterpriseContractPolicy
+		environment                *applicationapiv1alpha1.Environment
+		pipelineRun                *v1beta1.PipelineRun
+		release                    *v1alpha1.Release
+		releasePlan                *v1alpha1.ReleasePlan
+		releasePlanAdmission       *v1alpha1.ReleasePlanAdmission
+		releaseStrategy            *v1alpha1.ReleaseStrategy
+		snapshot                   *applicationapiv1alpha1.Snapshot
+		snapshotEnvironmentBinding *applicationapiv1alpha1.SnapshotEnvironmentBinding
+	)
+
+	AfterAll(func() {
+		deleteResources()
+	})
+
+	BeforeAll(func() {
+		createResources()
+
+		loader = NewLoader()
+	})
+
+	Context("When calling getObject", func() {
+		It("returns the requested resource if it exists", func() {
+			returnedApplication := &applicationapiv1alpha1.Application{}
+			err := getObject(application.Name, application.Namespace, k8sClient, ctx, returnedApplication)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(application.Spec).To(Equal(returnedApplication.Spec))
+		})
+
+		It("returns and error if the requested resource doesn't exist", func() {
+			returnedObject := &applicationapiv1alpha1.Application{}
+			err := getObject("non-existent-app", "non-existent-app", k8sClient, ctx, returnedObject)
+			Expect(err).To(HaveOccurred())
+			Expect(errors.IsNotFound(err)).To(BeTrue())
+			Expect(returnedObject).To(Equal(&applicationapiv1alpha1.Application{}))
+		})
+	})
+
+	Context("When calling GetActiveReleasePlanAdmission", func() {
+		It("returns an active release plan admission", func() {
+			returnedObject, err := loader.GetActiveReleasePlanAdmission(ctx, k8sClient, releasePlan)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(Equal(&v1alpha1.ReleasePlanAdmission{}))
+			Expect(returnedObject.Name).To(Equal(releasePlanAdmission.Name))
+		})
+
+		It("fails to return an active release plan admission if the target does not match", func() {
+			modifiedReleasePlan := releasePlan.DeepCopy()
```

Comment:
this should be deleted, similar to line 95, right?

Reply:
No. In this case I'm just passing an struct. In the test for line 95 I'm creating a resource in the cluster.